python/pyhail/dataset.py

```diff
@@ -49,6 +49,8 @@ def annotate_global_expr(self, condition):
 
         """
 
+        if isinstance(condition, list):
+            condition = ','.join(condition)
```

Comment:
These look good but can you update the types on condition?  Should be "str or list of str" now.